src/Macro.php

```diff
@@ -31,7 +31,26 @@ public function __construct(
      */
     protected function initPhpDoc($method)
     {
-        $this->phpdoc = new DocBlock($method);
+        $this->phpdoc = new DocBlock('/** */');
+
+        // Add macro parameters
+        foreach ($method->getParameters() as $parameter) {
+            $type = $parameter->hasType() ? $parameter->getType()->getName() : 'mixed';
+            $type .= $parameter->hasType() && $parameter->getType()->allowsNull() ? '|null' : '';
+
+            $name = $parameter->isVariadic() ? '...' : '';
+            $name .= '$' . $parameter->getName();
```

Comment:
Looks very clean but also duplicate code of what happens in `\Barryvdh\LaravelIdeHelper\Method::getParameters`

- the other method seems more feature-complete though I can't say if this is necessary
- but it's written in a way which makes it not easily re-usable AFAICS, i.e. you just can't call it.

Would you be interested comparing the the two approaches and see if yours is missing something? E.g. default values seems to be handled, etc.

----
Unifying this is topic for a different PR IMHO and shouldn't be done here.